src/Microsoft.Kusto.ServiceLayer/DataSource/Monitor/MonitorDataSource.cs

```diff
@@ -108,8 +108,31 @@ public override IEnumerable<DataSourceObjectMetadata> GetChildObjects(DataSource
 
             if (parentMetadata.MetadataType == DataSourceMetadataType.Cluster && includeSizeDetails)
             {
-                var child = _nodes[parentMetadata.Urn].FirstOrDefault();
-                return child == null ? Enumerable.Empty<DataSourceObjectMetadata>() : _nodes[child.Urn];
+                var tableGroups = _nodes[parentMetadata.Urn];
+
+                if (tableGroups == null)
+                {
+                    return Enumerable.Empty<DataSourceObjectMetadata>();
+                }
+
+                var results = new SortedList<string, DataSourceObjectMetadata>(StringComparer.OrdinalIgnoreCase);
+                foreach (var tableGroup in tableGroups)
```

Comment:
You could use some Linq statements here to make this more concise. Something to do for the future.